### ISSUE TITLE
Mouse Hook troubling again

### ISSUE THREAD

**Issue by oywino** [mzomparelli/zVirtualDesktop#249]
The Mouse Hook feature randomly stops working on my computer. Deselecting and re-selecting moue hook setting does nothing. Killing zVD fixes the problem - for a while (average 2 weeks).
Today, I discovered by chance that when the hook failed, the two side buttons on my mouse suddenly began working as back/forward buttons in Google Chrome (!)
I don't know if this info can be of any help in debugging the case. Why (and how) Chrome managed to overtake the mouse buttons, beats me. I haven't done anything. Chrome has always been my default browser.
My Chrome is Version 78.0.3904.87 (Official Build) (64-bit)

**Comment by mzomparelli** [comment]
Forward and Back are default browser behavior for those buttons and that simply implies that no hooks are handling it. Can you confirm what version of zVD you have as I have made further updates to this and may not have mentioned it to you. I have not released a version that notifies users yet. I think the last version I published is 1.0.93.6

**Comment by oywino** [comment]
I have 1.0.93.4
How can I upgrade? If I try - it tells me that I have the most recent version.

**Comment by mzomparelli** [comment]
Click the force update button in the lower-left of the settings window.

The Check for updates option only works when I publish a main version.

**Comment by oywino** [comment]
OK, I tried that but it tells me that the update was unsuccessful, and to contact you :-)
(Access to C:\Program Files (x86) was denied)

**Comment by mzomparelli** [comment]
What is your Win10 version? I will send you a link to download the update. It sounds like you manually moved zVD to your Program Files folder and zVD is not running as admin thus it cannot change files inside of Program Files.

Alternatively you could try closing zVD and running as admin to do the Force update. That should work. Another option is to move zVD to another folder and run from there. Since the settings are stored elsewhere you can run the exe from any location and use the same settings file.

**Comment by oywino** [comment]
Yes, I did move it to the Program Files (X86) folder. I felt that it was logical since afterall zVD is a 'program'. But now I moved it to the Document folder, and the FOrce update went well and my mouse hook is again back. I'll keep you posted when/if it fails again. Thanks for your good support. :)

**Comment by mzomparelli** [comment]
I didn't mean to imply it was illogical. It is very logical. It's also logical for Microsoft to prevent non-admin from making changes to that folder. :)

**Comment by mzomparelli** [comment]
Have you seen an improvement to this in the recent builds? I know I added code to try and maintain the mouse hook. Let me know. Thanks for your feedback.

**Comment by oywino** [comment]
I haven't had this problem for a long time now. It's been very stable.
Another new problem that I've had for the past 6 months though, is that regular left mouse clicks are suddenly not picked up by Windows. I have no reason to suspect zVD. I just thought I'd mention it.  I have not found any specific chain of events that lead up to this situation. I cannot reproduce it. It just suddenly happens.  To fix it, I press Alt-Ctrl-Del and select Task Manager. Then everything gets back to normal again - for a long time (weeks).

**Comment by mzomparelli** [comment]
So all you have to do is open Task Manager and your left clicks start working again. That sounds strange. How often does that occur?

**Comment by oywino** [comment]
That's right. I do not think that this has anything to do with Task Manager. It was just coincidental that I chose Task Manager. This is what happened the first time:  Because the mouse stopped  responding, I thought I had to restart the PC to get it going again. But before I did, I wanted to check what programs I had running - but there was no way to start Task Manager without a mouse. Then it occurred to me that Task Manager is one choice on the Alt-Ctrl-Del menu. And I knew I could select it by using the arrow keys and the enter key. So I did - and miraculously, when Task Manager was launched, the mouse woke up again.
So that's why I since that first day, continue to use Task Manager as my way of waking up the dead mouse. (Dead = left click only). Right click works all the time, and I can move it around on the screen. It tends to occur 2-3 times a month.  I have not found or discovered any pattern.

**Comment by mzomparelli** [comment]
That's odd that only the left click stops working. I will try to look for a cause for you. Maybe it's zVD maybe it isn't. I don't think it is, but I am intrigued to get to the bottom of it.

**Comment by oywino** [comment]
Also, what I have found to be a bit strange:
I have the following entry in registry: Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Run
Key Name: zVirtualDesktop
Key Value: "C:\Users\eBruker\Documents\zVirtualDesktop.exe"
But it fails!

But if I instead add a shortcut to my shell:startup folder, it works ok. Why?

**Comment by mzomparelli** [comment]
Yes, that registry entry is created when you check the run at startup option in zVD settings. Do you run zVD with Admin privileges? I know that the reg key doesn't like to start when zVD is configured to run as admin. Is "C:\Users\eBruker\Documents\zVirtualDesktop.exe" the correct location for your zVD executable?

**Comment by oywino** [comment]
Yes, I do log on to Windows with an Admin account (I that's what you mean).
And yes, "C:\Users\eBruker\Documents\zVirtualDesktop.exe" is the correct location.

**Comment by mzomparelli** [comment]
Not the account you log in with. I'm referring to how the application is launched. See the screenshots for two ways to run an app with admin privileges.

![image](https://user-images.githubusercontent.com/10564234/76550059-7e308080-645f-11ea-9f9c-f134440adab8.png)

![image](https://user-images.githubusercontent.com/10564234/76550100-8ee0f680-645f-11ea-8a26-159b48dea588.png)

**Comment by mzomparelli** [comment]
When you launch zVD do you get the Microsoft UAC prompt to allow the application to run?

**Comment by oywino** [comment]
No, never seen the UAC with zVD (never ever, in any version), and I have not made any changes to the settings of the executable.

**Comment by mzomparelli** [comment]
ok, I have no idea then why the registry is not starting it then. Since you put zVD in the startup folder and also have the reg key then how many zVD do you see in the Startup items in Task Manager? I feel like maybe the reg one got disabled in there.

![image](https://user-images.githubusercontent.com/10564234/76572638-a2ed1e00-6488-11ea-933c-df4e9b662a21.png)

**Comment by oywino** [comment]
Having both active simultaneously never resulted in more than one instance in Task Manager.
When I deleted the shortcut from the startup folder, zVD was never launched.
Adding the shortcut back to the startup folder, and everything was fine. 
The presence or absence from registry doesn't seem to have any impact at all.

**Comment by mzomparelli** [comment]
It won't ever allow a second instance because I explicitly handle that in the startup routine of zVD. I'll look into this more, but I'm not really sure what would cause this.
